craft_providers/bases/buildd.py

```diff
@@ -199,11 +202,8 @@ def _set_hostname(self, hostname: str) -> None:
             )
         valid_name = trimmed_name.group("valid_name")
```

Comment:
Can we replace this with
```
valid_name = name_with_valid_chars.strip("-")
if not valid_name:
    raise BaseConfigurationError(...)
```
?

Reply:
🤦 of course there was a more pythonic way to do this!  Simplified.